### PR TITLE
New tokenURLs + increase minimum bufferTime + updated validations

### PR DESCRIPTION
**Pull request by khtufteland**
commit e15d425aea4547a16000cdeac8f6a8f6948e95dc
Author: Kjetil Hope Tufteland <kjetil@searis.no>
Date:   Wed Sep 15 10:43:38 2021 +0200

    clarify_api: Update tokenURLs due to new domain

commit 4ee59eab989fbdb612f98bb8fc682d1c28c2053d
Author: Kjetil Hope Tufteland <kjetil@searis.no>
Date:   Mon Sep 13 14:49:56 2021 +0200

    clarify_insert: Set minimum bufferTime to 5 sec
    
    Some requests requires 2 seconds difference between SaveSignals and Insert.

commit fbb3ca9156be7fb97655ae6d3f781b28b969a0ca
Author: Kjetil Hope Tufteland <kjetil@searis.no>
Date:   Mon Sep 13 14:45:26 2021 +0200

    clarify_util: Update map[str][]str validators to not allow empty strings

commit 4559853cff81ce6716a8d14c9c095ea8b0d0d2b1
Author: Kjetil Hope Tufteland <kjetil@searis.no>
Date:   Tue Sep 7 23:42:05 2021 +0200

    README.md: Cache database is moved to folder, with many files

### CODE REVIEW

File: nodes/clarify_api.js [hunk 1]
```diff
@@ -60,10 +60,10 @@ module.exports = function (RED) {
       if (this.credentials.tokenUrl === undefined) {
         switch (url.host) {
           case 'api.clfy.io': // dev api (proxy)
-            this.credentials.tokenUrl = 'https://login.clarify.clfy.io/oauth/token';
+            this.credentials.tokenUrl = 'https://api.clfy.io/login/token';
             break;
-          case 'api.clarify.us': // prod api (proxy)
-            this.credentials.tokenUrl = 'https://login.clarify.us/oauth/token';
+          case 'api.clarify.io': // prod api (proxy)
```
Comment (bbergshaven) [inline nodes/clarify_api.js:65]: Make a default case that would work for both .us and .io (we will proxy both in the beginning)?

`default:
    this.credentials.tokenUrl = `${this.credentials.apiUrl}/login/token`
`

Comment (smyrman) [reply nodes/clarify_api.js:65]: I think it's actually
```js
this.credentials.tokenUrl = `${this.credentials.apiUrl}/oauth/toke`
```

Comment (smyrman) [reply nodes/clarify_api.js:65]: Can drop the switch-case all-together, if we are able to hold this off until we get the proxy for oauth/token up and running.

Comment (khtufteland) [reply nodes/clarify_api.js]: Commit is updated. I'll won't merge PR until proxy for oauth/token is up and running.

File: nodes/clarify_api.js [hunk 2]
```diff
@@ -56,24 +55,16 @@ module.exports = function (RED) {
         this.credentials.integrationId = this.credentials.overrideIntegrationId;
       }
 
-      let url = new URL(this.credentials.apiUrl);
+      // Strip slash from apiUrl
+      this.credentials.apiUrl = this.credentials.apiUrl.replace(/\/$/, '');
```
Comment (bbergshaven) [inline nodes/clarify_api.js:59]: Stripping slashes here leads to an invalid audience.

Comment (khtufteland) [reply nodes/clarify_api.js:63]: You're right. Didn't test it before pushing it. I've updated the code and added a short comment about it.

**Review by bbergshaven — APPROVED** [review]
Looks good.

**Pull request merged**